engine/src/main/scala/cromwell/engine/workflow/ExecutionStoreKey.scala

```diff
@@ -1,47 +1,32 @@
 package cromwell.engine.workflow
 
+import cromwell.engine.ExecutionStatus
+import cromwell.engine.backend.JobKey
 import cromwell.engine.finalcall.FinalCall
-import wdl4s._
-import cromwell.engine.{WorkflowDescriptor, ExecutionStatus}
 import cromwell.engine.workflow.WorkflowActor.ExecutionStore
+import wdl4s._
 
 import scala.language.postfixOps
 
-sealed trait ExecutionStoreKey {
+sealed trait ExecutionStoreKey extends JobKey {
```

Comment:
I think I managed to convince myself it was ok, saying that Backend can define a trait containing the stuff it needs to do its thing, and Engine can build on top of that to add its own internal sauce as it pleases..